templates/cluster/openstack-standalone-cp/templates/openstackmachinetemplate-controlplane.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: OpenStackMachineTemplate
+metadata:
+  name: {{ include "openstackmachinetemplate.controlplane.name" . }}
+spec:
+  template:
+    spec:
+      flavor: {{ .Values.controlplane.flavor }}
+      cloudName: {{ .Values.clusterIdentity.cloudName}}
+      image:
+        filter:
+          name: {{ .Values.controlplane.image.filter.name }}
+          {{- if .Values.controlplane.image.filter.tags }}
+          tags:
+            {{- range $tag := .Values.controlplane.image.filter.tags }}
+            - {{ $tag }}
+            {{- end }}
+          {{- end }}
+      {{- if gt (len .Values.controlplane.portOpts) 0 }}
```

Comment:
This check looks extra and you probably can just use `with` keyword here to be more consistent. Applicable to other `if gt` checks.

Reply:
if you use `with`, and let's say portOpts is defined like this:

```
portOpts: []
```

then in the rendered template, we get empty portOpts yaml. Using a `gt` keyword prevents this. Same for other vars - maps or list.

Reply:
>Is the portOpts: [] causing any issues?

already mentioned this. It will render empty portOpts in the final yaml.


>For securityGroups you have values defined, so it will never be empty (unless user explicitly delete defaults)

- it is an optional field
- yes, unless the user removes the default values, it won't be empty
- the gt will help handle even such a situation - it won't render empty blocks when its empty/non-existent in values.yaml

Reply:
>  It will render empty portOpts in the final yaml.
But is this causing any issues?

Reply:
since it is meaningless, it will be removed by kubernetes when it processes the yaml. But you don't want to handle it yourself if there is an option to do so?